lib/features/children/generosity_challenge/pages/generosity_challenge.dart

```diff
@@ -50,12 +50,21 @@ class _GenerosityChallengeState extends State<GenerosityChallenge>
     super.dispose();
   }
 
+  Future<void> _onCompleteChallenge() async {
+    await context.read<GenerosityChallengeCubit>().completeChallenge();
+    if (mounted) {
+      context.goNamed(Pages.childrenOverview.name);
```

Comment:
Can you get the navigation stack in order? I think this would just push the children overview page on top of the generosity challenge. But what we actually need is the current navigation stack cleared and then home and then on top of home the childrenoverview.

Bonus: if you can get the "download the kids4gift app screen" to go in between.